sdks/java/maven-archetypes/examples/build.gradle

```diff
@@ -55,6 +55,17 @@ task generateSources(type: Exec) {
   commandLine './generate-sources.sh'
 }
 
+// Add archetype testing/generation to existing GradleBuild PreCommit
+// https://issues.apache.org/jira/browse/BEAM-3256
+task archetypesTest(type: Exec) {
+  if (project.hasProperty("maven_home")) {
+    commandLine "${maven_home}/bin/mvn", 'clean', 'install'
+    environment "MAVEN_HOME", "${maven_home}"
+  } else {
+    commandLine 'mvn', 'clean', 'install'
```

Comment:
I'm investigating some [pre-commit failure](https://builds.apache.org/job/beam_PreCommit_Java_GradleBuild/5505/) on this task. I don't see anywhere as part of the execution that actually installs dependent maven artifacts in the local maven repository.

Is there anything to ensure that the maven command will only use artifacts generated as part of this build?

/cc @yifanzou @lukecwik

Reply:
Not part of this change. This change uses whichever mvn repo the developer/integration framework has which means that it will attempt to use the nightly snapshot.